pkg/tasks/alertmanager.go

```diff
@@ -329,6 +341,18 @@ func (t *AlertmanagerTask) destroy(ctx context.Context) error {
 		}
 	}
 
+	netpol, err := t.factory.AlertmanagerNetworkPolicy()
+	if err != nil {
+		return fmt.Errorf("initializing Alertmanager NetworkPolicy object failed: %w", err)
+	}
+
+	if netpol != nil {
+		err = t.client.DeleteNetworkPolicy(ctx, netpol)
```

Comment:
we can add a check that the NP disappears here https://github.com/openshift/cluster-monitoring-operator/blob/709ab51a727c8d2f8d2755d4e3c263021eaf5079/test/e2e/alertmanager_test.go#L679